src/script/widgets/ClassificationPanel.js

```diff
@@ -0,0 +1,317 @@
+                            //this.rulePanel.fireEvent("change", this.rulePanel, this.rulePanel.rule);
+                        },
+                        scope: this
+                    }
+                },
+                {   xtype: "gxp_colorfield",
+                    id: "choropleth_color_end",
+                    name: "color_end",
+                    fieldLabel: this.colorEndText,
+                    emptyText: OpenLayers.Renderer.defaultSymbolizer.strokeColor,
+                    value: this.endColor,
+                    defaultBackground: this.endColor,
+                    plugins: colorFieldPlugins,
+                    listeners: {
+                        valid: function(field) {
+                            this.rulePanel.rule[field.name] = field.getValue();
+                            //this.rulePanel.fireEvent("change", this.rulePanel, this.rulePanel.rule);
+                        },
+                        scope: this
+                    },
+                    scope: this
+                }
+
+            ]
+        });
+
+
+
+        var classNumSelector = new Ext.ux.form.SpinnerField({
+            fieldLabel: 'Classes',
+            id: "choropleth_classes",
+            minValue: 2,
+            name: 'intervals',
+            defaultValue: 5,
+            width: 110,
+            listeners: {
+                'change':function(spinner, value){
+                    this.rulePanel.rule['intervals'] = value;
+                    //this.rulePanel.fireEvent("change", this.rulePanel, this.rulePanel.rule);
+                },
+                scope: this
+            }
+        });
+
+        this.rulePanel.rule["intervals"] = classNumSelector.defaultValue;
+
+        var colorDropdown = new Ext.form.ComboBox({
+            id: 'choropleth_colorramp',
+            name: 'ramp',
+            fieldLabel: 'Color Ramp',
+            store:  new Ext.data.ArrayStore({
+                id: 0,
+                fields: [
+                    'colorramp',
+                    'label'
+                ],
+                data: [
+                    ['Blue',this.rampBlueText],
+                    ['Red', this.rampRedText],
+                    ['Orange', this.rampOrangeText],
+                    ['Jet', this.rampJetText],
+                    ['Gray', this.rampGrayText],
+                    ['Random', this.rampRandomText],
+                    ['Custom', this.rampCustomText]]
+            }),
+            mode: 'local',
+            width: 110,
+            displayField: "label",
+            valueField: "colorramp",
+            editable: false,
+            emptyText: this.selectColorText,
+            triggerAction: 'all',
+            disabled: false,
+            listeners: {
+                'select': function(cmb, data, idx) {
+                    //If Custom: display start, end, middle color picker;
+                    colorPanel.setVisible(cmb.value == "Custom");
```

Comment:
Can we make "Custom" a property of this class, so it doesn't have to get repeated all the time?